usaspending_api/disaster/management/sql/spark/disaster_covid19_file_d2_awards.py

```diff
@@ -0,0 +1,199 @@
+    EXTRACT (YEAR FROM (TO_DATE(latest_transaction.action_date)) + INTERVAL 3 months) AS award_latest_action_date_fiscal_year,
+    award_search.period_of_performance_start_date AS period_of_performance_start_date,
+    award_search.period_of_performance_current_end_date AS period_of_performance_current_end_date,
+    latest_transaction.awarding_agency_code AS awarding_agency_code,
+    latest_transaction.awarding_toptier_agency_name AS awarding_agency_name,
+    latest_transaction.awarding_sub_tier_agency_c AS awarding_sub_agency_code,
+    latest_transaction.awarding_subtier_agency_name AS awarding_sub_agency_name,
+    latest_transaction.awarding_office_code AS awarding_office_code,
+    latest_transaction.awarding_office_name AS awarding_office_name,
+    latest_transaction.funding_agency_code AS funding_agency_code,
+    latest_transaction.funding_toptier_agency_name AS funding_agency_name,
+    latest_transaction.funding_sub_tier_agency_co AS funding_sub_agency_code,
+    latest_transaction.funding_subtier_agency_name AS funding_sub_agency_name,
+    latest_transaction.funding_office_code AS funding_office_code,
+    latest_transaction.funding_office_name AS funding_office_name,
+    (SELECT CONCAT_WS(';', COLLECT_SET(U2.tas_rendering_label)) AS value FROM rpt.award_search U0 LEFT OUTER JOIN int.financial_accounts_by_awards U1 ON (U0.award_id = U1.award_id) LEFT OUTER JOIN global_temp.treasury_appropriation_account U2 ON (U1.treasury_account_id = U2.treasury_account_identifier) WHERE U0.award_id = (award_search.award_id) GROUP BY U0.award_id) AS treasury_accounts_funding_this_award,
+    (SELECT CONCAT_WS(';', COLLECT_SET(U3.federal_account_code)) AS value FROM rpt.award_search U0 LEFT OUTER JOIN int.financial_accounts_by_awards U1 ON (U0.award_id = U1.award_id) LEFT OUTER JOIN global_temp.treasury_appropriation_account U2 ON (U1.treasury_account_id = U2.treasury_account_identifier) LEFT OUTER JOIN global_temp.federal_account U3 ON (U2.federal_account_id = U3.id) WHERE U0.award_id = (award_search.award_id) GROUP BY U0.award_id) AS federal_accounts_funding_this_award,
+    (SELECT CONCAT_WS(';', COLLECT_SET(CONCAT(U2.object_class, ':', U2.object_class_name))) FROM rpt.award_search U0 LEFT OUTER JOIN int.financial_accounts_by_awards U1 ON (U0.award_id = U1.award_id) INNER JOIN global_temp.object_class U2 ON (U1.object_class_id = U2.id) WHERE U0.award_id = (award_search.award_id) and U1.object_class_id IS NOT NULL GROUP BY U0.award_id) AS object_classes_funding_this_award,
+    (SELECT CONCAT_WS(';', COLLECT_SET(CONCAT(U2.program_activity_code, ':', U2.program_activity_name))) FROM rpt.award_search U0 LEFT OUTER JOIN int.financial_accounts_by_awards U1 ON (U0.award_id = U1.award_id) INNER JOIN global_temp.ref_program_activity U2 ON (U1.program_activity_id = U2.id) WHERE U0.award_id = (award_search.award_id) and U1.program_activity_id IS NOT NULL GROUP BY U0.award_id) AS program_activities_funding_this_award,
```

Comment:
I believe you'll want to add `array_sort(...)` or `SORT_ARRAY(...)` to these multi-valued results. While not strictly necessary it helps with testing using the diff tool to compare data from place to place and not lead to false failures due to out-of-order items. Also provides some consistency for consumers of the data.

Reply:
We should add this function to the PSQL too, right? Otherwise, spark sql would be sorted while the psql results wouldn't be.

Reply:
FIXED - Added sorting to these lines where they appear in all the Covid-19 download queries. PSQL and SparkSQL